src/PointEvalHandler.jl

```diff
@@ -89,54 +97,85 @@ function _get_cellcoords(points::AbstractVector{Vec{dim,T}}, grid::AbstractGrid,
     return cells, local_coords
 end
 
-# check if point is inside a cell based on physical coordinate
-function point_in_cell(geom_interpol::Interpolation{shape}, cell_coordinates, global_coordinate) where {shape}
-    converged, x_local = find_local_coordinate(geom_interpol, cell_coordinates, global_coordinate)
-    if converged
-        return _check_isoparametric_boundaries(shape, x_local), x_local
-    else
-        return false, x_local
-    end
-end
-
 # check if point is inside a cell based on isoparametric coordinate
-function _check_isoparametric_boundaries(::Type{RefHypercube{dim}}, x_local::Vec{dim, T}) where {dim, T}
-    tol = sqrt(eps(T))
-    # All in the range [-1, 1]
-    return all(x -> abs(x) - 1 < tol, x_local)
+function check_isoparametric_boundaries(::Type{RefHypercube{dim}}, x_local::Vec{dim, T}, tol) where {dim, T}
+    # All in the range [-1, 1]^dim
+    return all(x -> abs(x) - 1 ≤ tol, x_local)
 end
 
 # check if point is inside a cell based on isoparametric coordinate
-function _check_isoparametric_boundaries(::Type{RefSimplex{dim}}, x_local::Vec{dim, T}) where {dim, T}
-    tol = sqrt(eps(T))
+function check_isoparametric_boundaries(::Type{RefSimplex{dim}}, x_local::Vec{dim, T}, tol) where {dim, T}
     # Positive and below the plane 1 - ξx - ξy - ξz
     return all(x -> x > -tol, x_local) && sum(x_local) - 1 < tol
 end
 
+cellcenter(::Type{<:RefHypercube{dim}}, _::Type{T}) where {dim, T} = zero(Vec{dim, T})
+cellcenter(::Type{<:RefSimplex{dim}}, _::Type{T}) where {dim, T} = Vec{dim, T}((ntuple(d->1/3, dim)))
+
+_solve_helper(A::Tensor{2,dim}, b::Vec{dim}) where {dim} = inv(A) ⋅ b
+_solve_helper(A::SMatrix{idim, odim}, b::Vec{idim,T}) where {odim, idim, T} = Vec{odim,T}(pinv(A) * b)
+
 # See https://discourse.julialang.org/t/finding-the-value-of-a-field-at-a-spatial-location-in-juafem/38975/2
-# TODO: should we make iteration params optional keyword arguments?
-function find_local_coordinate(interpolation, cell_coordinates::Vector{<:Vec{dim}}, global_coordinate::Vec{dim}; tol_norm = 1e-10) where dim
+function find_local_coordinate(interpolation::Interpolation{ref_shape}, cell_coordinates::Vector{<:Vec{sdim}}, global_coordinate::Vec{sdim}, strategy::NewtonLineSearchPointFinder; warn::Bool = false) where {sdim, ref_shape}
```

Comment:
Just detail, but I think we only use `refshape` or `RefShape` in other places in the codebase. (Update required below as well)
```suggestion
function find_local_coordinate(interpolation::Interpolation{refshape}, cell_coordinates::Vector{<:Vec{sdim}}, global_coordinate::Vec{sdim}, strategy::NewtonLineSearchPointFinder; warn::Bool = false) where {sdim, refshape}
```

Reply:
Okay we got all kinds of combinations.
```
src/interpolations.jl
2:    Interpolation{ref_shape, order}()
4:Abstract type for interpolations defined on `ref_shape`
```
Is there an easy way to check such conventions in a pre-commit hook?

Reply:
Haha, ok, then nvm. 
I guess there would be some way via the typos in #865, but not sure if it is worth the effort.